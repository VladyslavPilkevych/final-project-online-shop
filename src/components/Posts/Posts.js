/* eslint-disable max-len */
import React from 'react';
import styles from './Posts.module.scss';

function Posts() {
  return (
    <div className={styles.posts}>
      <h1 className={styles.postsArticle}>Follow us on Instagram for News, Offers & More</h1>
      <div className={styles.postsWrapper}>
        <div className={styles.postItem}>
          <img src="/images/instagram-images/instagram-image1.png" className="post-image" alt="post-img" />
          <p className={styles.postText}>If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...</p>
          <p className={styles.postDate}>01.09.2020</p>
        </div>
        <div className={styles.postItem}>
          <img src="/images/instagram-images/instagram-image2.png" className="post-image" alt="post-img" />
          <p className={styles.postText}>If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...</p>
          <p className={styles.postDate}>01.09.2020</p>
        </div>
        <div className={styles.postItem}>
          <img src="/images/instagram-images/instagram-image3.png" className="post-image" alt="post-img" />
          <p className={styles.postText}>If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...</p>
          <p className={styles.postDate}>01.09.2020</p>
        </div>
        <div className={styles.postItem}>
          <img src="/images/instagram-images/instagram-image4.png" className="post-image" alt="post-img" />
          <p className={styles.postText}>If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...</p>
          <p className={styles.postDate}>01.09.2020</p>
        </div>
        <div className={styles.postItem}>
          <img src="/images/instagram-images/instagram-image5.png" className="post-image" alt="post-img" />
          <p className={styles.postText}>If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...</p>
          <p className={styles.postDate}>01.09.2020</p>
        </div>
        <div className={styles.postItem}>
          <img src="/images/instagram-images/instagram-image6.png" className="post-image" alt="post-img" />
          <p className={styles.postText}>If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...</p>
          <p className={styles.postDate}>01.09.2020</p>
        </div>
        <div className={styles.postItem}>
          <img src="/images/instagram-images/instagram-image7.png" className="post-image" alt="post-img" />
          <p className={styles.postText}>If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...</p>
          <p className={styles.postDate}>01.09.2020</p>
        </div>
        <div className={styles.postItem}>
          <img src="/images/instagram-images/instagram-image8.png" className="post-image" alt="post-img" />
          <p className={styles.postText}>If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...</p>
          <p className={styles.postDate}>01.09.2020</p>
        </div>
        <div className={styles.postItem}>
          <img src="/images/instagram-images/instagram-image9.png" className="post-image" alt="post-img" />
          <p className={styles.postText}>If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...</p>
          <p className={styles.postDate}>01.09.2020</p>
        </div>

      </div>
    </div>
  );
}

export default Posts;
