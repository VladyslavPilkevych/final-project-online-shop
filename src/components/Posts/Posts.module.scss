@mixin fonts($font-family, $size, $color, $bold) {
    font-family: $font-family, sans-serif;
    font-weight: $bold;
    font-size: $size;
    color: $color;
}

.posts {
    max-width: 1398px;
    margin: 0 auto;
}

.postsWrapper {
    margin-top: 20px;
    column-gap: 8px;
    row-gap: 20px;
    display: grid;
    grid-template-columns: repeat(6, 1fr);
    justify-items: center;
}

.postsArticle {
    @include fonts("Poppins", 22px, #000000, 600);
    line-height: 33px;
}


.postItem {
   display: flex;
   flex-direction: column;
   justify-content: center;
   align-items: center;
   text-align: center;
   max-width: 225px;
}

.postText {
    @include fonts("Poppins", 12px, #000000, 400);
    margin-top: 10px;
    line-height: 18px;
}

.postDate {
    @include fonts("Poppins", 10px, #A2A6B0, 400);
    margin-top: 10px;
    line-height: 15px;
}

@media screen and (min-device-width: 1157px) and (max-device-width: 1393px) {
    .postsWrapper {
        display: grid;
        grid-template-columns: repeat(5, 1fr);
        justify-items: center;
    }

    .postsArticle {
        margin-left: 22px;
    }
}

@media screen and (min-device-width: 925px) and (max-device-width: 1156px) {
    .postsWrapper {
        display: grid;
        grid-template-columns: repeat(4, 1fr);
        justify-items: center;
    }

    .postsArticle {
        margin-left: 25px;
    }
}

@media screen and (min-device-width: 768px) and (max-device-width: 925px) {
    .postsWrapper {
        display: grid;
        grid-template-columns: repeat(3, 1fr);
        justify-items: center;
    }

    .postsArticle {
        margin-left: 34px;
    }
}

@media screen and (min-device-width: 375px) and (max-device-width: 767px) {
    .posts {
       display: none;
    }
}
