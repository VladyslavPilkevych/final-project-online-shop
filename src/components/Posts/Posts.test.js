/* eslint-disable react/react-in-jsx-scope */
import { render } from '@testing-library/react';
import Posts from './Posts';

describe('Snapshot posts should render', () => {
  test('Snapshot posts is rendered', () => {
    const { asFragment } = render(<Posts />);
    expect(asFragment()).toMatchSnapshot();
  });
});