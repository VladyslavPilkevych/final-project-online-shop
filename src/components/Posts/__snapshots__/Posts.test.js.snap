// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot posts should render Snapshot posts is rendered 1`] = `
<DocumentFragment>
  <div
    class="posts"
  >
    <h1
      class="postsArticle"
    >
      Follow us on Instagram for News, Offers & More
    </h1>
    <div
      class="postsWrapper"
    >
      <div
        class="postItem"
      >
        <img
          alt="post-img"
          class="post-image"
          src="instagram-image1.png"
        />
        <p
          class="postText"
        >
          If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...
        </p>
        <p
          class="postDate"
        >
          01.09.2020
        </p>
      </div>
      <div
        class="postItem"
      >
        <img
          alt="post-img"
          class="post-image"
          src="instagram-image2.png"
        />
        <p
          class="postText"
        >
          If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...
        </p>
        <p
          class="postDate"
        >
          01.09.2020
        </p>
      </div>
      <div
        class="postItem"
      >
        <img
          alt="post-img"
          class="post-image"
          src="instagram-image3.png"
        />
        <p
          class="postText"
        >
          If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...
        </p>
        <p
          class="postDate"
        >
          01.09.2020
        </p>
      </div>
      <div
        class="postItem"
      >
        <img
          alt="post-img"
          class="post-image"
          src="instagram-image4.png"
        />
        <p
          class="postText"
        >
          If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...
        </p>
        <p
          class="postDate"
        >
          01.09.2020
        </p>
      </div>
      <div
        class="postItem"
      >
        <img
          alt="post-img"
          class="post-image"
          src="instagram-image5.png"
        />
        <p
          class="postText"
        >
          If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...
        </p>
        <p
          class="postDate"
        >
          01.09.2020
        </p>
      </div>
      <div
        class="postItem"
      >
        <img
          alt="post-img"
          class="post-image"
          src="instagram-image6.png"
        />
        <p
          class="postText"
        >
          If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...
        </p>
        <p
          class="postDate"
        >
          01.09.2020
        </p>
      </div>
      <div
        class="postItem"
      >
        <img
          alt="post-img"
          class="post-image"
          src="instagram-image7.png"
        />
        <p
          class="postText"
        >
          If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...
        </p>
        <p
          class="postDate"
        >
          01.09.2020
        </p>
      </div>
      <div
        class="postItem"
      >
        <img
          alt="post-img"
          class="post-image"
          src="instagram-image8.png"
        />
        <p
          class="postText"
        >
          If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...
        </p>
        <p
          class="postDate"
        >
          01.09.2020
        </p>
      </div>
      <div
        class="postItem"
      >
        <img
          alt="post-img"
          class="post-image"
          src="instagram-image9.png"
        />
        <p
          class="postText"
        >
          If you’ve recently made a desktop PC or laptop purchase, you might want to consider adding peripherals to enhance your home office setup, your gaming rig, or your business workspace...
        </p>
        <p
          class="postDate"
        >
          01.09.2020
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
